User facing mcp integration documentation (#118)
trento-project/docs: trento/adoc/trento-mcp-install-kubernetes.adoc

include::product-attributes.adoc[]

[[sec-trento-mcp-server-k8s]]
==== {k8s} {tr_mcp} deployment
:revdate: 2025-11-17

include::partials/kubernetes-placeholders.adoc[]

===== Enable the {tr_mcp}

> When installing the complete {trserver} using {helm}, the {tr_mcp} is disabled by default. Enable it by passing `--set trento-mcp-server.enabled=true`:

Reviewer: ```suggestion
When installing {trserver} following the instructions in <<sec-trento-k8s-deployment>>, the {tr_mcp} is disabled by default. Enable it by passing `--set trento-mcp-server.enabled=true`:
```

====
[source,bash]
----
helm upgrade --install trento-server oci://registry.suse.com/trento/trento-server \

>   --set trento-web.trentoWebOrigin=TRENTO_SERVER_HOSTNAME \

Reviewer: ```suggestion
  --set global.trentoWeb.origin==TRENTO_SERVER_HOSTNAME \
```

  $(kubectl get pods --namespace default -l app.kubernetes.io/name=mcp-server -o jsonpath="{.items[0].metadata.name}") \
  8080:8080
----
+
[source,bash]
----
# Liveness endpoint:
curl http://localhost:8080/livez
----
+
[source,bash]
----
# Readiness endpoint:
curl http://localhost:8080/readyz
----
+
Example output:
+
[source,bash]
----
# Liveness:
{"info":{"name":"trento-mcp-server","version":"0.1.0"},"status":"up"}

# Readiness:
{"status":"up","details":{"mcp-server":{"status":"up","timestamp":"2025-10-09T12:11:09.528898849Z"},"wanda-api":{"status":"up","timestamp":"2025-10-09T12:11:09.544855047Z"},"web-api":{"status":"up","timestamp":"2025-10-09T12:11:09.544855047Z"}}}
----

===== {tr_mcp} {helm} configuration options

The {tr_mcp} {helm} chart supports various configuration options:

* <<mcp-customize-ingress-path,Customize Ingress Path>>
* <<mcp-disabling-tag-filtering,Disabling Tag Filtering>>
* <<mcp-adjust-log-verbosity,Adjust Log Verbosity>>
* <<mcp-adjust-resource-limits,Adjust Resource Limits>>
* <<mcp-disabling-health-check-probes,Disabling Health Check Probes>>

[[mcp-customize-ingress-path]]
====== Customize Ingress Path

By default, ingress is enabled.

> To customize the ingress configuration:

Reviewer: ```suggestion
To customize the ingress configuration in a basic K3s installation:
```

====
[source,bash]
----
helm upgrade --install trento-server oci://registry.suse.com/trento/trento-server \
  --set trento-web.trentoWebOrigin=TRENTO_SERVER_HOSTNAME \
  --set trento-web.adminUser.password=ADMIN_PASSWORD \
  --set trento-mcp-server.enabled=true \

>   --set trento-mcp-server.ingress.enabled=true \

Author: Yeah it is but It’s clearer to readers :) 
We can remove if you want

  --set trento-mcp-server.ingress.hosts[0].host=TRENTO_SERVER_HOSTNAME \
  --set trento-mcp-server.ingress.hosts[0].paths[0].path=/mcp-server-trento \
  --set trento-mcp-server.ingress.hosts[0].paths[0].pathType=ImplementationSpecific
----
====

The {tr_mcp} endpoint will be: `https://TRENTO_SERVER_HOSTNAME/mcp-server-trento/mcp`

[[mcp-disabling-tag-filtering]]
====== Disabling Tag Filtering

By default, only operations tagged with `MCP` are exposed. To expose all operations and remove tag filtering:

====
[source,bash]
----
helm upgrade --install trento-server oci://registry.suse.com/trento/trento-server \
  --set trento-web.trentoWebOrigin=TRENTO_SERVER_HOSTNAME \
  --set trento-web.adminUser.password=ADMIN_PASSWORD \
  --set trento-mcp-server.enabled=true \
  --set trento-mcp-server.mcpServer.tagFilter=[]
----
====

[NOTE]
====
You can also filter by different tags by setting `--set trento-mcp-server.mcpServer.tagFilter[0]=YourTag`.
====

[[mcp-adjust-log-verbosity]]
====== Adjust Log Verbosity

The default log level is `info`. Adjust it for debugging:

====
[source,bash]
----
helm upgrade --install trento-server oci://registry.suse.com/trento/trento-server \

>   --set trento-web.trentoWebOrigin=TRENTO_SERVER_HOSTNAME \

Reviewer: ```suggestion
  --set global.trentoWeb.origin=TRENTO_SERVER_HOSTNAME \
```

  --set trento-web.adminUser.password=ADMIN_PASSWORD \
  --set trento-mcp-server.enabled=true \
  --set trento-mcp-server.mcpServer.verbosity=debug
----
====

[[mcp-adjust-resource-limits]]
====== Adjust Resource Limits

For production deployments with different resource requirements:

====
[source,bash]
----
helm upgrade --install trento-server oci://registry.suse.com/trento/trento-server \

>   --set trento-web.trentoWebOrigin=TRENTO_SERVER_HOSTNAME \

Reviewer: ```suggestion
  --set global.trentoWeb.origin=TRENTO_SERVER_HOSTNAME \
```
